### PR TITLE
Add functions to work with lists of arrays

### PR DESCRIPTION
**Pull request by alvp**

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -74,6 +74,76 @@ def icm(beta: float, v1: np.ndarray, v2: np.ndarray):
...
+
+
+def f(operation: str, v1: np.ndarray, v2: np.ndarray):
+    """Calculates the chosen operation between two vectors
+
+    :param operation: A string with the possible values for the operations
+    :type operation: str
+    :param v1: Embedding vector 1
+    :type v1: np.ndarray
+    :param v2: Embedding vector 1
```
Comment (aitordiaz) [inline src/utils.py:89]: Please, fix this description

File: src/utils.py [hunk 2]
```diff
@@ -74,6 +74,76 @@ def icm(beta: float, v1: np.ndarray, v2: np.ndarray):
...
+    :type operation: str
+    :param embeddings_list: A list of embeddings
+    :type embeddings_list: List[np.array]
+    :return: Numpy array resulting of composition function
+    :rtype: np.ndarray
+    """
+    return get_composite_vector(operation, embeddings_list, r2l=False)
+
+
+def ind_rl(operation: str, embeddings_list: List[np.array]):
```
Comment (aitordiaz) [inline src/utils.py:139]: Are these two functions necessary?

**Review by aitordiaz — CHANGES_REQUESTED** [review]

File: src/utils.py [hunk 3]
```diff
@@ -74,6 +74,76 @@ def icm(beta: float, v1: np.ndarray, v2: np.ndarray):
...
+      embeddings_list = embeddings_list[::-1]
+    if embeddings_list:
+      v1 = embeddings_list.pop(0)
+    while embeddings_list:
+      v2 = embeddings_list.pop(0)
+      v1 = f(operation, v1, v2)
+    return v1
+
+
+def ind_lr(operation: str, embeddings_list: List[np.array]):
```
Comment (kasia-postdata) [inline src/utils.py:126]: I would like to have one function to compute any composition function from it. Something like:

def composition_function(composition_function_name: str, embedding_list: List[np.array])

where name of the composition function can be: 'sum', 'avg', 'avg_lr', 'avg_rl', 'ind_lr', 'ind_rl', 'joint_lr', 'joint_rl',

Comment (kasia-postdata) [reply src/utils.py:126]: or multiple function with the names 'sum', 'avg', 'avg_lr', 'avg_rl', 'ind_lr', 'ind_rl', 'joint_lr', 'joint_rl'
I don't mind.

But I would like to match the proper function just by one parameter - the name

File: src/utils.py [hunk 4]
```diff
@@ -74,6 +74,76 @@ def icm(beta: float, v1: np.ndarray, v2: np.ndarray):
...
+    """Applies the f function from right to left
+
+    :param operation: A string with the possible values for the operations
+    :type operation: str
+    :param embeddings_list: A list of embeddings
+    :type embeddings_list: List[np.array]
+    :return: Numpy array resulting of composition function
+    :rtype: np.ndarray
+    """
+    return get_composite_vector(operation, embeddings_list, r2l=True)
```
Comment (kasia-postdata) [inline src/utils.py:149]: Also, I can't see the standard average function implemented here.

File: src/utils.py (hunk 3 above)
Comment (aitordiaz) [reply src/utils.py:126]: Ok, understood 👍 Thanks for the review, I'm working on it.

File: src/utils.py (hunk 4 above)
Comment (aitordiaz) [reply src/utils.py]: Done!

**Comment by aitordiaz** [comment]
To call `composition_function`:

File: src/utils.py (hunk 3 above)
Comment (alvp) [reply src/utils.py]: We have taken the first approach, `composition_function(composition_function_name: str, embedding_list: List[np.array])`. 
The only difference is that "**joint_lr**" and "**joint_rl**" are now called "**jnt_lr**" and "**jnt_rl**"

**Review by kasia-postdata — APPROVED** [review]
Perfect! Thx

**Pull request merged**